engine/src/main/scala/cromwell/engine/workflow/ShadowWorkflowActor.scala

```diff
@@ -1,19 +1,166 @@
+  /**
+    * Responses from the ShadowWorkflowActor
+    */
+  sealed trait ShadowWorkflowActorResponse
+  case class ShadowWorkflowSucceeded(outputs: WorkflowOutputs) extends ShadowWorkflowActorResponse
+  case class ShadowWorkflowFailed(reason: Throwable) extends Exception with ShadowWorkflowActorResponse
+
+  /**
+    * States for the ShadowWorkflowActor FSM
+    */
+  sealed trait ShadowWorkflowActorState { def terminal = false }
+  sealed trait ShadowWorkflowActorTerminalState extends ShadowWorkflowActorState { override def terminal = true}
+
+  /**
+    * Waiting for a Start or Restart command.
+    */
+  case object Unstarted extends ShadowWorkflowActorState
+
+  /**
+    * The WorkflowActor is created. It now needs to validate and create its WorkflowDescriptor
+    */
+  case object MaterializingWorkflowDescriptor extends ShadowWorkflowActorState
+
+  /**
+    * The WorkflowActor has a descriptor. It now needs to create its set of WorkflowBackendActors
+    */
+  case object InitializingWorkflowBackendActors extends ShadowWorkflowActorState
+
+  /**
+    * The WorkflowActor has a descriptor and a set of backends. It now needs to execute the hell out of its jobs.
+    */
+  case object ExecutingWorkflow extends ShadowWorkflowActorState
+
+  /**
+    * The WorkflowActor has completed. So we're now finalizing whatever needs to be finalized on the backends
+    */
+  case object FinalizingWorkflow extends ShadowWorkflowActorState
+
+  /**
+    * We're done.
+    */
+  case object WorkflowSuccessful extends ShadowWorkflowActorTerminalState
+
+  /**
+    * We're done. But we were aborted in the middle of the lifecycle.
+    */
+  case object WorkflowAborted extends ShadowWorkflowActorTerminalState
+
+  case class ShadowWorkflowActorData(backendAssignments: Map[Call, String])
+
+  /**
+    * Mode in which the workflow should be started:
+    */
+  sealed trait StartMode
+  case object StartNewWorkflow extends StartMode
+  case object RestartExistingWorkflow extends StartMode
+
+  def props(workflowId: WorkflowId, startMode: StartMode, wdlSource: WorkflowSourceFiles): Props = Props(new ShadowWorkflowActor(workflowId, startMode, wdlSource))
 }
 
 /**
-  * Class that will [eventually] orchestrate a single workflow.
+  * Class that orchestrates a single workflow... Shadow style.
   */
-class ShadowWorkflowActor extends Actor with ActorLogging{
+class ShadowWorkflowActor(workflowId: WorkflowId, startMode: StartMode, wdlSource: WorkflowSourceFiles) extends LoggingFSM[ShadowWorkflowActorState, ShadowWorkflowActorData] with ActorLogging{
+
+  private val tag = s"$workflowId-${this.getClass.getSimpleName}"
 
-  private val tag = this.getClass.getSimpleName
+  startWith(Unstarted, ShadowWorkflowActorData(Map.empty))
+
+  val decider: PartialFunction[Throwable, Directive] = {
+    case _: Throwable => Escalate
+  }
+  override def supervisorStrategy: SupervisorStrategy =
+    OneForOneStrategy()(decider.orElse(SupervisorStrategy.defaultStrategy.decider))
```

Comment:
So this means... "If an actor that I'm supervising dies, escalate the exception to my supervisor"?

If I'm understanding this correctly, the decider that you've defined doesn't need an `orElse`, so why can't we just write `OneForOneStrategy()(decider)`?